.changeset/rare-days-punch.md

```diff
@@ -0,0 +1,5 @@
+---
+'@web3-ui/hooks': patch
+---
+
+Add check for user connection during contract init in useContract
```

Comment:
I think this would probably give a little extra context:

```suggestion
`useContract` now initializes the contract only after the user has connected their wallet. This helps prevent errors from calling contracts without a connection.
```